clients/roscpp/src/libros/io.cpp

```diff
@@ -380,6 +380,26 @@ pollfd_vector_ptr poll_sockets(int epfd, socket_pollfd *fds, nfds_t nfds, int ti
  * @return int : 1 on connected, 0 on connecting, -1 on failure.
  */
 int is_async_connected(socket_fd_t &socket, int &err) {
+#ifdef HAVE_EPOLL
+    int nfds = 1;
+    int epfd = create_socket_watcher();
+    add_socket_to_watcher(epfd, socket);
+    set_events_on_socket(epfd, socket, EPOLLIN | EPOLLOUT);
+    struct epoll_event ev[nfds];
+    int fd_cnt = ::epoll_wait(epfd, ev, nfds, 0);
+    close_socket_watcher(epfd);
+    if (fd_cnt < 0) {
+      if (errno != EINTR) {
+        ROS_ERROR("Error in epoll_wait! %s", strerror(errno));
+        err = errno;
+        return -1;
+      }
+    } else if (fd_cnt == 0) {
+      return 0;
```

Comment:
The existing code sets `err` to `0` on success in this case. Should this branch do the same?

Reply:
Done